core/strings/strings.odin

```diff
@@ -466,58 +830,142 @@ _split :: proc(s_, sep: string, sep_save, n_: int, allocator := context.allocato
-	Returned strings are substrings of 's'.
-	```
-	s := "aaa.bbb.ccc.ddd.eee" // 5 parts
-	ss := split(s, ".")
-	fmt.println(ss)            // [aaa, bbb, ccc, ddd, eee]
-	```
+Splits a string into parts based on a separator.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**  
+- s: The string to split.
+- sep: The separator string used to split the input string.
+- allocator: (default is context.allocator).
+
+Example:
+
+	import "core:fmt"
+	import "core:strings"
+
+	strings_split_example :: proc() {
+		s := "aaa.bbb.ccc.ddd.eee"    // 5 parts
+		ss := strings.split(s, ".")
+		fmt.println(ss)
+	}
+
+Output:
+
+	["aaa", "bbb", "ccc", "ddd", "eee"]
+
+**Returns**  A slice of strings, each representing a part of the split string.
 */
 split :: proc(s, sep: string, allocator := context.allocator) -> []string {
 	return _split(s, sep, 0, -1, allocator)
 }
-
 /*
-	Splits a string into a total of 'n' parts, based on a separator.
-	Returns fewer parts if there wasn't enough occurrences of the separator.
-	Returned strings are substrings of 's'.
-	```
-	s := "aaa.bbb.ccc.ddd.eee" // 5 parts present
-	ss := split_n(s, ".", 3)   // total of 3 wanted
-	fmt.println(ss)            // [aaa, bbb, ccc.ddd.eee]
-	```
+Splits a string into parts based on a separator. If n < count of seperators, the remainder of the string is returned in the last entry.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**  
+- s: The string to split.
+- sep: The separator string used to split the input string.
+- allocator: (default is context.allocator)
+
+Example:
+
+	import "core:fmt"
+	import "core:strings"
+
+	strings_split_n_example :: proc() {
+		s := "aaa.bbb.ccc.ddd.eee"  // 5 parts present
+		ss := strings.split_n(s, ".",3) // total of 3 wanted
+		fmt.println(ss)
+	}
+
+Output:
+
+	["aaa", "bbb", "ccc.ddd.eee"]
+
+**Returns**  A slice of strings, each representing a part of the split string.
 */
 split_n :: proc(s, sep: string, n: int, allocator := context.allocator) -> []string {
 	return _split(s, sep, 0, n, allocator)
 }
-
 /*
-	splits the string `s` after the seperator string `sep` appears
-	returns the slice of split strings allocated using `allocator`
+Splits a string into parts after the separator, retaining it in the substrings.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**  
+- s: The string to split.
+- sep: The separator string used to split the input string.
+- allocator: (Optional) The allocator used for allocation (default is context.allocator).
+
+Example:
+
+	import "core:fmt"
+	import "core:strings"
+
+	strings_split_after_example :: proc() {
+		a := "aaa.bbb.ccc.ddd.eee"         // 5 parts
+		aa := strings.split_after(a, ".")
+		fmt.println(aa)
+	}
+
+Output:
 
-	a := "aaa.bbb.ccc.ddd.eee"
-	aa := strings.split_after(a, ".")
-	fmt.eprintln(aa) // [aaa., bbb., ccc., ddd., eee]
+	["aaa.", "bbb.", "ccc.", "ddd.", "eee"]
+
+**Returns**  A slice of strings, each representing a part of the split string after the separator.
```

Comment:
Maybe this?

**Returns** A slice of strings, split after the separator.  
NOTE: Allocation occurs for the slice, the splits are all views into the original string.

Reply:
i like views, makes it less confusing.